### ISSUE TITLE
On Firefox 59.0 Smart Referer prevents playing Facebook videos

### ISSUE THREAD

**Issue by AntonioRedondo** [meh/smart-referer#96]
Hi meh,

It looks like Smart Referer prevents videos being played on Facebook. This was not the case just a week ago. Facebook has made some change and know it requires the referer being sent in order to play videos. As soon as I disable the add-on videos play normally. I've added the exceptions `*.fbcdn.net` and `*.facebook.com` to the add-on options but videos still don't play.

Am I missing something?

**Comment by l0gik** [comment]
I can verify this issue, appearing recently (FWIW it appears to be CORS-related).  However, unlike OP I was able to restore video playback by adding `*.facebook.com` as the source, and `video.*.fna.fbcdn.net` as the destination to the Exceptions list.  NB: I haven't mapped out the FB CDN servers or anything, so that destination might be too specific depending on where the video is hosted.

@AntonioRedondo: if the Exceptions workaround doesn't do it for you, have you ensured that other extensions have been ruled out? I had to do some extension toggling to isolate this particular problem to Smart Referer.

**Comment by alexander255** [comment]
Thanks @0gik! I cannot verify this, but if @AntonioRedondo can test this I think we can add it to the whitelist anyway.

**Comment by AntonioRedondo** [comment]
Hi guys, thanks for the quick reply. I just tried adding `*.facebook.com` as the source, and `video.*.fna.fbcdn.net` as the destination, but it didn't work. I also tried `*.fna.fbcdn.net`. Still no luck. However, if the destinations is only `*.fbcdn.net` then videos reproduce again.

**Comment by vertigo220** [comment]
I just ran into this on Waterfox 56.2.0. I added *.facebook.com as the source and *.fbcdn.net as the destination and it worked. I did it this way, before even coming here, because I don't see the point in restricting it, since they're both facebook, and doing so will only cause it to break if/when it doesn't use the same exact server (e.g., mine was video-ort2-1.xx.fbcdn.net). Using *.fbcdn.net should prevent issues in all cases. This should probably be added to the built-in whitelist, since as I said, they're both facebook anyway, so there's not really a reason not to.
